lib/logging/log.js

```diff
@@ -0,0 +1,415 @@
+ * Write a log entry with a severity of "CRITICAL".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.critical({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.critical = function(entry, labels, callback) {
+  var entries = Log.assignSeverityToEntries_(entry, 'CRITICAL');
+  this.insert(entries, labels, callback);
+};
+
+/**
+ * Write a log entry with a severity of "DEBUG".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.debug({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.debug = function(entry, labels, callback) {
+  this.insert(Log.assignSeverityToEntries_(entry, 'DEBUG'), labels, callback);
+};
+
+/**
+ * Write a log entry with a severity of "EMERGENCY".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.emergency({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.emergency = function(entry, labels, callback) {
+  var entries = Log.assignSeverityToEntries_(entry, 'EMERGENCY');
+  this.insert(entries, labels, callback);
+};
+
+/**
+ * Write a log entry with a severity of "ERROR".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.error({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.error = function(entry, labels, callback) {
+  this.insert(Log.assignSeverityToEntries_(entry, 'ERROR'), labels, callback);
+};
+
+/**
+ * Write a log entry with a severity of "INFO".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.info({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.info = function(entry, labels, callback) {
+  this.insert(Log.assignSeverityToEntries_(entry, 'INFO'), labels, callback);
+};
+
+/**
+ * Write log entries to Cloud Logging.
+ *
+ * @resource [projects.logs.entries.write API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.logs.entries/write}
+ *
+ * @param {object|object[]} entry - A log entry, or array of entries, to write.
```

Comment:
Maybe it's worth adding that batching entries is highly recommended, as the is a QPS limit.